### PR TITLE
Re-enabling tests that were flaky

### PR DESCRIPTION
**Pull request by Piinks**
## Description

A lot of golden file web tests were disabled due to flakiness. https://github.com/flutter/flutter/issues/47003 was tracking this flakiness. It has been marked as resolved, so I am re-enabling these tests. If they still flake, we will have a recent reproduction. :)

## Related Issues

https://github.com/flutter/flutter/issues/47003

## Tests

are a go! 🚀 

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by skia-gold** [comment]
Gold has detected one or more untriaged digests on patchset 1.
Please triage them at https://flutter-gold.skia.org/search?issue=51666.

**Comment by Piinks** [comment]
> Gold has detected one or more untriaged digests on patchset 1.
> Please triage them at https://flutter-gold.skia.org/search?issue=51666.

Some of these tests were previously not enabled, and chrome updates probably made slight changes here and there. Triaging...

**Comment by chingjun** [comment]
LGTM but curious about two things:

Some of the tests you enabled here, were not enabled previously because I couldn't get some of them running, just want to make sure that you are able to get them running this time.

Also there are 4 more tests in `bottom_navigation_bar_test.dart` and 1 in `opacity_test.dart` not enabled in here, is there any reason why they are not enabled?

**Comment by Piinks** [comment]
I wanted to try enabling them all and seeing what the results were, either reproducing the flake or identifying ones that weren't working.
So far they all look ok except for this one (the web version is on the left, and doesn't appear to render the nav bar):

Screen Shot 2020-02-28 at 3 39 08 PM

Do you have an idea why the bottom nav bar wouldn't work? 

> Also there are 4 more tests in bottom_navigation_bar_test.dart and 1 in opacity_test.dart not enabled in here, is there any reason why they are not enabled?

I must have missed those! Thanks! I will see if those present issues like the bottom nav bar above.

Should I file bugs for each one that doesn't work out?

**Comment by Piinks** [comment]
> Also there are 4 more tests in `bottom_navigation_bar_test.dart` and 1 in `opacity_test.dart` not enabled in here, is there any reason why they are not enabled?

I checked, and these skipped tests aren't golden file tests. They don't have comments either saying why they are skipped. Perhaps running an audit of `skip: isBrowser` would be a good idea for another change. There are a lot of skipped tests that aren't clear why they are being skipped.

File: packages/flutter/test/widgets/shader_mask_test.dart [hunk 1]
```diff
@@ -108,5 +108,5 @@ void main() {
       find.byType(RepaintBoundary),
       matchesGoldenFile('shader_mask.bounds.matches_top_left.png'),
     );
-  }, skip: isBrowser);
+  });
```
Comment (Piinks) [inline packages/flutter/test/widgets/shader_mask_test.dart]:
```suggestion
  }, skip: isBrowser); // https://github.com/flutter/flutter/issues/44152
```

**Comment by skia-gold** [comment]
Gold has detected one or more untriaged digests on patchset 2.
Please triage them at https://flutter-gold.skia.org/search?issue=51666.

**Comment by Piinks** [comment]
> Do you have an idea why the bottom nav bar wouldn't work?

Sorry, it's bottom app bar. Not nav bar.

**Comment by Piinks** [comment]
I am going to re-skip the golden file test for the BottomAppBar, as it appears that it is actually broken. Filed: https://github.com/flutter/flutter/issues/51675

**Comment by Piinks** [comment]
Looks like the flake was reproduced as well on one test run: https://cirrus-ci.com/task/6066690064646144?command=main#L1244

I will re-open the flake.

File: packages/flutter/test/material/bottom_app_bar_test.dart [hunk 2]
```diff
@@ -79,7 +79,7 @@ void main() {
       find.byKey(key),
       matchesGoldenFile('bottom_app_bar.custom_shape.2.png'),
     );
-  }, skip: isBrowser);
+  });
```
Comment (Piinks) [inline packages/flutter/test/material/bottom_app_bar_test.dart]:
```suggestion
  }, skip: isBrowser); // https://github.com/flutter/flutter/issues/51675
```

**Comment by skia-gold** [comment]
Gold has detected one or more untriaged digests on patchset 3.
Please triage them at https://flutter-gold.skia.org/search?issue=51666.

**Comment by yjbanov** [comment]
We're not ready to enable web golden tests yet. Sorry for misunderstanding.

**Comment by Piinks** [comment]
I will close this PR since the flake has not been resolved.

**Pull request closed without merging**